src/api/notes.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/notes"
+
+	wh "github.com/skycoin/skycoin/src/util/http"
+)
+
+var (
+	// ErrorWrongTxID appears when TransactionID has wrong format
+	ErrorWrongTxID = "wrong 'txid'"
+	// ErrorBadParams appears when note obj isn't complete
+	ErrorBadParams = "bad parameters"
+)
+
+// URI: /api/v2/notes
+// Method: GET
+// Content-Type: application/json
+// Body: -
+// Response:
+//      200 - ok, returns all notes
+func getAllNotesHandler(gateway Gatewayer) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			wh.Error405(w)
+			return
+		}
+
+		savedNotes := gateway.GetAllNotes()
+
+		wh.SendJSONOr500(logger, w, savedNotes)
+	}
+}
+
+// URI: /api/v2/note
+// Method: POST, GET, DELETE
+// Content-Type: application/json
+// Body: { "txid": "<Transaction ID>" }
+// Response:
+//      400 - bad parameters
+//      200 - POST: returns added Note
+//			- GET: return note by Transaction ID
+//			- DELETE: removes note by Transaction ID
+func noteHandler(gateway Gatewayer) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost {
+			var note notes.Note
```

Comment:
Should check that the request content type is `application/json` (there are examples of this in other v2 endpoints)